server/src/repositories/asset.repository.ts

```diff
@@ -576,12 +592,26 @@ export class AssetRepository implements IAssetRepository {
       .$if(!!options.userIds, (qb) => qb.where('assets.ownerId', '=', anyUuid(options.userIds!)))
       .$if(options.isArchived !== undefined, (qb) => qb.where('assets.isArchived', '=', options.isArchived!))
       .$if(options.isFavorite !== undefined, (qb) => qb.where('assets.isFavorite', '=', options.isFavorite!))
-      .$if(!!options.withStacked, (qb) => withStack(qb, { assets: true, count: false })) // TODO: optimize this; it's a huge performance hit
+      .$if(!!options.withStacked, (qb) =>
+        withStack(qb, { assets: true, count: false, withDeleted: false, withArchived: false }),
+      ) // TODO: optimize this; it's a huge performance hit
       .$if(!!options.assetType, (qb) => qb.where('assets.type', '=', options.assetType!))
       .$if(options.isDuplicate !== undefined, (qb) =>
         qb.where('assets.duplicateId', options.isDuplicate ? 'is not' : 'is', null),
       )
       .$if(!!options.isTrashed, (qb) => qb.where('assets.status', '!=', AssetStatus.DELETED))
+      .$if(!!options.tagId, (qb) =>
+        qb.where((eb) =>
+          eb.exists(
+            eb
+              .selectFrom('tags')
```

Comment:
I don't think this needs to do anything with the tag table. You just need to make sure it exists in tags_assets and that only needs the closure table afaik. Unless you are selecting the tag name, you shouldn't even need to query that table for stuff like this.